skt/__init__.py

```diff
@@ -405,10 +405,27 @@ def merge_patchwork_patch(self, uri):
                           patchinfo.get("name").replace(',', ';')))
 
     def merge_patch_file(self, path):
-        try:
-            self.git_cmd("am", path)
-        except subprocess.CalledProcessError:
+        if not os.path.isfile(path):
+            raise Exception("Patch %s not found" % os.path.basename(path))
+        abspath = os.path.abspath(path)
+
+        gam = subprocess.Popen(
+            ["git", "am", abspath],
+            cwd=self.wdir,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            env=dict(os.environ, **{'LC_ALL': 'C'})
+        )
+
+        (stdout, stderr) = gam.communicate()
```

Comment:
Since we don't need to pass any input for communicate and only grab `stdout`, this can be implemented much simpler by `subprocess.check_output` and then saving the mergelog from exception's `output` attribute.

Reply:
Could you elaborate? Because I see `git_cmd` already uses `subprocess.check_output. Do you have the way to get `output` attribute from this function without modifying?